src/bindgen/ir/generic_path.rs

```diff
@@ -10,6 +10,50 @@ use crate::bindgen::ir::{ConstExpr, Path, Type};
 use crate::bindgen::utilities::IterHelpers;
 use crate::bindgen::writer::{Source, SourceWriter};
 
+// Struct that serves as key for resolving associated types
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct AssocTypeId{
+    pub ty: Box<Type>,
+    pub trait_: Path,
+    pub ident: Path,
+}
+
+impl AssocTypeId{
+    
+    pub fn new(path: &syn::Path, qself: &Option<syn::QSelf>) -> Result<Self, String>{
+        
+        let self_type = if let Some(qself) = qself{
+            &qself.ty
+        } else {
+            return Err(String::from("Type is not fully-qualified."))
+        };
+
+        // Extract type
+        let ir_type_opt = Type::load(&self_type)?;
+        let ir_type = if let Some(ty) = ir_type_opt{
+            ty
+        } else{
+            return Err(String::from("Valid but empty type"))
+        };
+
+        let path_len = path.segments.len();
+        // Theoretically not possible if qself is present
+        if path_len < 2{
+            return Err(String::from("Trait not found in type"))
+        }
+
+        let ident = path.segments.last().unwrap().ident.to_string();
+        let trait_ident = path.segments[path_len - 2].ident.to_string();
```

Comment:
I find this very weird. Shouldn't you do:

```suggestion
        let trait_ident = path.segments.last().ident.to_string();
```